src/main/kotlin/proofs/Theorem6Conj2.kt

```diff
@@ -0,0 +1,30 @@
+package proofs
+
+import parsing.Conjunction
+import parsing.System
+
+class Theorem6Conj2: Proof {
+    //HSCC.Theorem6.part2, locally consistent S, U and T: (U <= S) and (U <= T) implies U <= (S && T)
+    override fun search(component: System, ctx: ProofSearcher.IterationContext) {
+        if (component.thisRefines.size > 1) {
+            val refines = HashSet<System>()
+            refines.addAll(component.thisRefines)
+            for (lhs in refines) {
+                for (rhs in refines) {
+                    if (!lhs.sameAs(rhs)){
+                        var newComp : System = Conjunction(lhs, rhs)
+                        newComp = ctx.addNewComponent(newComp)
+                        if(newComp.refinesThis.add(component)) {
+                            ctx.setDirty(newComp)
+                        }
+                        if(component.thisRefines.add(newComp)) {
+                            ctx.setDirty(component)
+                        }
+                    }
+                }
+            }
+
+        }
```

Comment:
I might be misunderstanding the explanation of this theorem on line 7. But isnt one of the requirements for U <= (S && T) to hold that U, S and T are all have to be locally consistent? If so we should add local consistency as a property on a system and include it in the check here.

Reply:
If they were not locally consistent, we would not have the relation, no? The refinement check would fail if either side were not locally consistent, so doesn't the relation imply local consistency?